include/albatross/src/core/prediction.hpp

```diff
@@ -218,5 +218,12 @@ auto get_prediction(const ModelType &model, const FitType &fit,
   return Prediction<ModelType, FeatureType, FitType>(model, fit, features);
 }
 
+template <typename ModelType, typename FeatureType, typename FitType>
+auto get_prediction_reference(const ModelType &model, const FitType &fit,
```

Comment:
I guess `PredictionReference` is an object with references stored inside of itself? So you aren't returning a ` T &` or `T &&` type here?

Reply:
Right, `PredictionReference` just changes the internal stored types for `model_` and `fit_` to be const references: https://github.com/swift-nav/albatross/blob/master/include/albatross/src/core/declarations.hpp#L51-L53